File: archives/C#/0383. Ransom Note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        Dictionary<char,int> magazineDict=new Dictionary<char,int>();
        foreach(char c in magazine){
            if(!magazineDict.ContainsKey(c)){
                magazineDict[c]=1;
            }
            else{
                magazineDict[c]+=1;
            }
        }
        bool flag=true;
        foreach(char c in ransomNote){
            if(!magazineDict.ContainsKey(c)){
                flag=false;
                break;
            }
            else{
                if(magazineDict[c]==0){
                    flag=false;
                    break;
                }
                else{
                    magazineDict[c]-=1;
                }
            }
        }
        return flag;
    }
}

File: archives/20190519python/0383. Ransom Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        magazine_dict={}
        for c in magazine:
            if c not in magazine_dict:
                magazine_dict[c]=1
            else:
                magazine_dict[c]+=1
                
        for c in ransomNote:
            if c not in magazine_dict:
                return False
            else:
                if magazine_dict[c]==0:
                    return False
                magazine_dict[c]-=1
        return True
